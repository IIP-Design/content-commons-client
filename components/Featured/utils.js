import { capitalizeFirst } from 'lib/utils';
import { normalizeItem } from 'lib/elastic/parser';
import { postTypeAggRequest } from 'lib/elastic/api';

/**
 * Get a string of up to three categories from a content item
 * @param {Object} item Normalized data for a give content item
 * @returns {string} Lowercase list of category names separated by ·
 */
export const getCategories = item => {
  const cats = item?.categories?.slice( 0, 3 ).reduce( ( acc, cat, index, arr ) => {
    const c = acc + cat.name.toLowerCase();

    return index < arr.length - 1 && index < 2 ? `${c} · ` : c;
  }, '' );

  return cats;
};

/**
 * Updates the the featured context
 * @param {Promise[]} array Group of promises that resolve the recents properties
 * @param {dispatch} dispatch A reducer dispatch function
 */
export const getFeatured = async ( array, dispatch ) => {
  const resArr = await Promise.all( array ).catch( err => {
    dispatch( {
      type: 'LOAD_FEATURED_FAILED',
    } );
  } );

  console.dir( resArr );

  const priorities = {};
  const recents = {};
  let items;

  if ( resArr ) {
    resArr.forEach( res => {
      if ( res?.data?.hits?.hits ) {
        items = res.data.hits.hits.map( item => normalizeItem( item, res.locale ) );

        switch ( res.component ) {
          case 'priorities':
            priorities[res.term] = items;
            break;
          case 'packages':
          case 'recents':
            recents[res.postType] = items;
            break;
          default:
            break;
        }
      }
    } );

    dispatch( {
      type: 'LOAD_FEATURED_SUCCESS',
      payload: {
        priorities,
        recents,
      },
    } );
  }
};


/**
 *
 * @param {dispatch} dispatch A reducer dispatch function
 * @param {Object} user User date object
 */
export const loadPostTypes = async ( dispatch, user ) => {
  dispatch( { type: 'LOAD_POST_TYPES_PENDING' } );

  let response;

  try {
    response = await postTypeAggRequest( user );
  } catch ( err ) {
    dispatch( { type: 'LOAD_POST_TYPES_FAILED' } );

    return;
  }

  if ( response?.aggregations?.postType?.buckets ) {
    const { buckets } = response.aggregations.postType;

    const filtered = buckets.filter( type => type.key !== 'courses' && type.key !== 'page' && type.key !== 'package' );

    const payload = filtered.map( type => {
      let displayName = capitalizeFirst( type.key );

      if ( type.key === 'post' ) displayName = 'Article';
      if ( type.key === 'document' ) displayName = 'Press Releases and Guidance';

      return {
        key: type.key,
        display_name: displayName,
        count: type.doc_count,
      };
    } );

    dispatch( {
      type: 'LOAD_POST_TYPES_SUCCESS',
      payload,
    } );
  } else {
    dispatch( {
      type: 'LOAD_POST_TYPES_SUCCESS',
      payload: [],
    } );
  }
};
