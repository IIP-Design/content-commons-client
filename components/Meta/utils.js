/**
 * Check if content exists for a given meta property
 * If not, filters it from the meta object
 *
 * @param {Object} meta All possible meta tags
 * @returns {Object} Only those meta tags for which there is content
 */
const filterEmpty = meta => {
  const filteredMeta = {};
  const metaArr = Object.entries( meta );

  metaArr.forEach( item => {
    if ( item[1] ) {
      const property = item[0];
      const value = item[1];

      filteredMeta[property] = value;
    }
  } );

  return filteredMeta;
};

/**
 * Converts the content item object into a metavalues array using the checkAndPush method
 *
 * @param {object} item Object of metadata for a given content item
 * @param {string} url The URL of the page to be rendered
 * @returns {array} Array of { property: content } key pairs
 */
export const getOgTags = ( item, url ) => {
  const {
    selectedLanguageUnit,
    title,
    type,
  } = item;

  const isVideo = type && type === 'video';

  /* TEMP vars for Document use case */
  const trimContent = item.content && item.content.rawText && `${item.content.rawText.substring( 0, 152 )}...`;
  const description = item.description || trimContent;

  const videoSrc = selectedLanguageUnit && selectedLanguageUnit.source && selectedLanguageUnit.source[0]
    ? selectedLanguageUnit.source[0]
    : {};
  const streamSrc = videoSrc.stream && videoSrc.stream.link ? videoSrc.stream.link : '';

  const meta = {
    'og:type': isVideo ? 'video.other' : 'article',
    'og:url': url,
    'twitter:card': isVideo ? 'player' : 'summary',
    'og:description': description,
    'og:title': title,
    'og:video:url': isVideo ? streamSrc : '',
    'twitter:player': isVideo ? streamSrc : '',
    'twitter:player:height': isVideo ? '196px' : '',
    'twitter:player:width': isVideo ? '350px' : '',
  };

  return filterEmpty( meta );
};
