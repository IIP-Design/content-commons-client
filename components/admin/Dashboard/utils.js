import { getProjectsType } from 'lib/graphql/util';
import {
  DELETE_GRAPHIC_PROJECT_MUTATION,
  TEAM_GRAPHIC_PROJECTS_QUERY,
  TEAM_GRAPHIC_PROJECTS_COUNT_QUERY,
  GRAPHIC_PROJECT_IMAGE_FILES_QUERY,
  UNPUBLISH_GRAPHIC_PROJECT_MUTATION,
  UPDATE_GRAPHIC_STATUS_MUTATION,
  GRAPHIC_PROJECTS_META_QUERY,
} from 'lib/graphql/queries/graphic';
import {
  DELETE_PACKAGE_MUTATION,
  PACKAGE_FILES_QUERY,
  TEAM_PACKAGES_COUNT_QUERY,
  TEAM_PACKAGES_QUERY,
  UNPUBLISH_PACKAGE_MUTATION,
  UPDATE_PACKAGE_STATUS_MUTATION,
  PACKAGES_META_QUERY,
} from 'lib/graphql/queries/package';
import {
  DELETE_VIDEO_PROJECT_MUTATION,
  TEAM_VIDEO_PROJECTS_COUNT_QUERY,
  TEAM_VIDEO_PROJECTS_QUERY,
  UNPUBLISH_VIDEO_PROJECT_MUTATION,
  UPDATE_VIDEO_STATUS_MUTATION,
  VIDEO_PROJECT_FILES_QUERY,
  VIDEO_PROJECTS_META_QUERY,
} from 'lib/graphql/queries/video';
import { TEAM_PLAYBOOKS_QUERY, TEAM_PLAYBOOKS_COUNT_QUERY } from 'lib/graphql/queries/playbook';

/**
 * Returns the project type based off of the content type available to a team
 *
 * @param {Object} team team data received from GraphQL
 */
export const getContentTypesFromTeam = team => {
  const contentTypes = team?.contentTypes ? team.contentTypes : '';

  const type = getProjectsType( contentTypes );

  return type || null;
};

/**
 * Returns the expected queries depending on what content types a team has access to
 *
 * @param {Object} team team data received from GraphQL
 * @returns {Object} list of relevant queries
 */
export const setQueries = team => {
  const queries = {};

  switch ( getContentTypesFromTeam( team ) ) {
    case 'graphicProjects':
      queries.content = TEAM_GRAPHIC_PROJECTS_QUERY;
      queries.count = TEAM_GRAPHIC_PROJECTS_COUNT_QUERY;
      queries.files = GRAPHIC_PROJECT_IMAGE_FILES_QUERY;
      queries.metaContent = GRAPHIC_PROJECTS_META_QUERY;
      queries.remove = DELETE_GRAPHIC_PROJECT_MUTATION;
      queries.status = UPDATE_GRAPHIC_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_GRAPHIC_PROJECT_MUTATION;

      return queries;
    case 'videoProjects':
      queries.content = TEAM_VIDEO_PROJECTS_QUERY;
      queries.count = TEAM_VIDEO_PROJECTS_COUNT_QUERY;
      queries.files = VIDEO_PROJECT_FILES_QUERY;
      queries.metaContent = VIDEO_PROJECTS_META_QUERY;
      queries.remove = DELETE_VIDEO_PROJECT_MUTATION;
      queries.status = UPDATE_VIDEO_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_VIDEO_PROJECT_MUTATION;

      return queries;
    case 'packages':
      queries.content = TEAM_PACKAGES_QUERY;
      queries.count = TEAM_PACKAGES_COUNT_QUERY;
      queries.files = PACKAGE_FILES_QUERY;
      queries.metaContent = PACKAGES_META_QUERY;
      queries.remove = DELETE_PACKAGE_MUTATION;
      queries.status = UPDATE_PACKAGE_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_PACKAGE_MUTATION;

      return queries;

    case 'playbooks':
      queries.content = TEAM_PLAYBOOKS_QUERY;
      queries.count = TEAM_PLAYBOOKS_COUNT_QUERY;
      // The following queries are placeholders and for the wrong content type
      // As such, these operations will not work properly
      queries.files = PACKAGE_FILES_QUERY;
      queries.metaContent = PACKAGES_META_QUERY;
      queries.remove = DELETE_PACKAGE_MUTATION;
      queries.status = UPDATE_PACKAGE_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_PACKAGE_MUTATION;

      return queries;
    default:
      return queries;
  }
};
