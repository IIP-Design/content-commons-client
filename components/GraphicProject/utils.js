import { getTransformedLangTaxArray, getCount } from 'lib/utils';

const structureLangObj = item => ( {
  language: {
    language_code: item.language.languageCode,
    text_direction: item.language.textDirection,
    locale: item.language.locale,
    display_name: item.language.displayName,
    native_name: item.language.nativeName,
  },
} );

export const normalizeGraphicProjectByAPI = ( { file, useGraphQl = false } ) => {
  const graphicObj = {
    id: file.id || '',
    type: 'graphic',
    site: file.site || '',
    title: file.title || '',
    projectType: file.projectType || '',
    published: file.published || '',
    modified: file.modified || '',
    owner: file.owner || '',
    desc: file.desc || '',
    descInternal: file.descInternal || '',
    copyright: file.copyright || '',
    images: file.images || [],
    supportFiles: file.supportFiles || [],
    categories: file.categories || '',
  };

  // Elastic API
  if ( !useGraphQl ) {
    const imagesWithFileNameProp = file.images.map( img => ( {
      ...img,
      filename: img.filename,
    } ) );

    const supportFilesWithFileNameProp = Array.isArray( file.supportFiles ) && file.supportFiles.map( supFile => {
      const filenameFromSrc = supFile.url.slice( supFile.url.lastIndexOf( '/' ) + 1 );

      return {
        ...supFile,
        filename: filenameFromSrc,
      };
    } );

    const esObj = {
      images: imagesWithFileNameProp || [],
      supportFiles: supportFilesWithFileNameProp || [],
    };

    return { ...graphicObj, ...esObj };
  }

  // GraphQL API
  if ( useGraphQl ) {
    const structuredImages = file.images.map( img => {
      const { dimensions } = img;

      return {
        ...img,
        ...structureLangObj( img ),
        url: img.signedUrl,
        width: dimensions.width,
        height: dimensions.height,
      };
    } );

    const structuredSupportFiles = Array.isArray( file.supportFiles ) && file.supportFiles.map( supFile => ( {
      ...supFile,
      ...structureLangObj( supFile ),
      srcUrl: supFile.signedUrl,
    } ) );

    const gqlObj = {
      alt: file.alt || '',
      projectType: file.type || '',
      published: file.publishedAt || '',
      modified: file.updatedAt || '',
      owner: file.team?.name || '',
      desc: file.descPublic || '',
      images: structuredImages || [],
      supportFiles: structuredSupportFiles || [],
      categories: getTransformedLangTaxArray( file.categories ) || [],
    };

    return { ...graphicObj, ...gqlObj };
  }

  return graphicObj;
};

// Use Twitter graphics as default for display otherwise whatever graphic image is available
export const filterGraphicImgs = images => {
  const containsTwitterImgs = images.some( img => img.social.includes( 'Twitter' ) );

  if ( containsTwitterImgs ) {
    return images.filter( img => img.social.includes( 'Twitter' ) );
  }

  return images;
};

export const getGraphicImgsBySocial = ( images, platform, useGraphQl = false ) => {
  if ( !getCount( images ) ) return [];

  const filteredImgs = images.reduce( ( allImgs, img ) => {
    let platformImgs = [];
    let condition = img.social.includes( platform );

    if ( useGraphQl ) {
      platformImgs = img.social.filter( s => s.name === platform );
      condition = getCount( platformImgs );
    }

    if ( condition ) {
      allImgs.push( img );
    }

    return allImgs;
  }, [] );

  return getCount( filteredImgs ) ? filteredImgs : images;
};
