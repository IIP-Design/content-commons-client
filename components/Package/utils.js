import { getTransformedLangTaxArray } from 'lib/utils';
import { packageDocumentsRequest } from 'lib/elastic/api';
import { getDataFromHits } from 'lib/elastic/parser';
import DosSeal from 'static/images/dos_seal.svg';

/*
 * Normalize Document type data structure from both APIs
 *
 */
export const normalizeDocumentItemByAPI = ( { file, useGraphQl = false } ) => {
  const documentObj = {
    type: 'document',
    id: file.id || '',
    published: file.published || '',
    modified: file.modified || '',
    author: file.author || '',
    owner: file.owner || '',
    site: file.site || '',
    title: file.title || '',
    excerpt: file.excerpt || '',
    content: file.content || '',
    logo: DosSeal || '',
    language: file.language,
    documentUrl: file.url || '',
    documentUse: file.use || '',
    tags: file.tags || [],
  };

  if ( useGraphQl ) {
    const graphFields = {
      created: file.createdAt || '',
      published: file.publishedAt || '',
      modified: file.updatedAt || '',
      author: file.author ? `${file.author.firstname} ${file.author.lastname}` : '',
      owner: file.team?.name || '',
      documentUrl: file.url || '',
      documentUse: file?.use?.name || '',
      excerpt: file?.excerpt || '',
      tags: getTransformedLangTaxArray( file.tags ) || [],
    };

    return { ...documentObj, ...graphFields };
  }

  return documentObj;
};

/*
 * Request documents for a package from Elasticsearch
 * @param docs array of objects w shape { id, item }
 */
export const getElasticPkgDocs = async ( documents, user ) => {
  if ( user && user.id !== 'public' ) {
    try {
      const docIds = documents.map( doc => doc.id );
      const responseHits = await packageDocumentsRequest( docIds, user );
      const docs = getDataFromHits( responseHits )?.map( hit => hit._source );

      return docs;
    } catch ( error ) {
      console.log( error );

      return error;
    }
  }
};
