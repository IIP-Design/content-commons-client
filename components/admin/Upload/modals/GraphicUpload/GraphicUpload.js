import React, { useEffect, useState } from 'react';
// import dynamic from 'next/dynamic';
import PropTypes from 'prop-types';
import { useApolloClient } from '@apollo/react-hooks';
import { useRouter } from 'next/router';

import { normalize } from 'lib/graphql/normalizers/graphic';
import { useFileStateManager } from 'lib/hooks/useFileStateManager';
import { serializeFile } from 'lib/utils';

import EditFileGrid from 'components/admin/EditFileGrid/EditFileGrid';
import GraphicStyleDropdown from 'components/admin/dropdowns/GraphicStyleDropdown/GraphicStyleDropdown';
import LanguageDropdown from 'components/admin/dropdowns/LanguageDropdown/LanguageDropdown';
import SocialPlatformDropdown from 'components/admin/dropdowns/SocialPlatformDropdown/SocialPlatformDropdown';
import IncludeRequiredFileMsg from 'components/admin/Upload/modals/IncludeRequiredFileMsg/IncludeRequiredFileMsg';

import styles from './GraphicUpload.module.scss';

const GraphicUpload = ( { files, closeModal } ) => {
  const client = useApolloClient();
  const router = useRouter();

  const IMAGE_EXTS = '.png,.jpg,.jpeg,.gif';
  const EDITABLE_EXTS = '.psd,.ai,.ae';
  const OTHER_EXTS = '.pdf,.doc,.docx,.ttf,.otf';

  // What files does this modal accept?
  const ALLOWED_FILES = `${IMAGE_EXTS},${EDITABLE_EXTS},${OTHER_EXTS}`;

  // Defines how may "tabs" appear and which components, e.g dropdowns that they container
  const screens = [
    [
      {
        label: 'Language',
        name: 'language',
        component: LanguageDropdown,
        // What files does this dropdown accept?
        // do NOT include .ttf, .otf files in lang dropdown
        allowedFiles: `${IMAGE_EXTS},.pdf,.doc,.docx,`,
        props: { search: true },
      },
    ],
    [
      {
        label: 'Style',
        name: 'style',
        component: GraphicStyleDropdown,
        allowedFiles: IMAGE_EXTS,
      },
      {
        label: 'Platform',
        name: 'social',
        component: SocialPlatformDropdown,
        allowedFiles: IMAGE_EXTS,
        props: { closeOnChange: true },
      },
    ],
  ];

  // Set state with files normalized for the graphic content type
  const { state, dispatch } = useFileStateManager( null, normalize( files ) );
  const [hasGraphicFiles, setHasGraphicFiles] = useState( false );
  const [showRequiredFilesMsg, setShowRequiredFilesMsg] = useState( false );

  const getGraphicFiles = () => {
    if ( state?.files ) {
      return state.files.filter( file => {
        const { input: { type }, styleSelection } = file;

        const isRequiredFileType = type.includes( 'gif' ) || type.includes( 'jpeg' ) || type.includes( 'png' );
        // Do not allow required file types with 'Clean' style to be added as graphic files
        const isReqdFileWithCleanStyle = isRequiredFileType && styleSelection === 'Clean';

        return isRequiredFileType && !isReqdFileWithCleanStyle;
      } );
    }

    return [];
  };

  useEffect( () => {
    const graphicFiles = getGraphicFiles();
    const count = graphicFiles.length;

    setHasGraphicFiles( !!count );
  }, [state.files] );

  /**
   * Serialize File Objects, write local object to Apollo cache, go to graphic edit page
   */
  const create = async () => {
    // display required files message
    if ( !hasGraphicFiles ) {
      setShowRequiredFilesMsg( true );

      return null;
    }

    // Loop thru files, serialize File Object and create object
    const fileList = await Promise.all(
      state.files.map( async file => {
        // File Object must be serialized for Apollo cache store
        const dataUrl = await serializeFile( file.input );

        return {
          __typename: 'LocalImageFile',
          ...file,
          input: {
            __typename: 'LocalInputFile',
            dataUrl,
            name: file.input.name,
            size: file.input.size,
          },
        };
      } ),
    ).catch( err => {
      // console.log( `Unable to process files ${err.toString()}` );
    } );

    // If there are images, Write files to apollo cache
    if ( fileList?.length ) {
      client.writeData( {
        data: {
          localGraphicProject: {
            __typename: 'LocalGraphicProject',
            files: fileList,
          },
        },
      } );

      // Go to GraphicEdit page
      router.push( {
        pathname: '/admin/project',
        query: {
          content: 'graphic',
          action: 'edit',
        },
      } );
    }
  };

  return (
    <div className={ styles.container }>
      <h1 style={ { fontSize: '1rem' } }>SOCIAL MEDIA GRAPHIC PROJECT</h1>
      <p className={ styles.files }>{ `Preparing ${state.files.length} files for upload...` }</p>

      <EditFileGrid
        files={ state.files }
        screens={ screens }
        allowedFiles={ ALLOWED_FILES }
        duplicateConfirm
        removeConfirm
        cancelConfirm
        showHeader
        showCompletionTracker
        continueBtn="Continue"
        onCancel={ closeModal }
        onContinue={ create }
        onAdd={ _files => dispatch( { type: 'ADD', files: normalize( _files ) } ) }
        onUpdate={ data => dispatch( { type: 'UPDATE', data } ) }
        onRemove={ file => dispatch( { type: 'REMOVE', fileId: file.id } ) }
      />

      <IncludeRequiredFileMsg
        msg="Please include at least one graphic file."
        includeRequiredFileMsg={ showRequiredFilesMsg }
        setIncludeRequiredFileMsg={ setShowRequiredFilesMsg }
      />
    </div>
  );
};

GraphicUpload.propTypes = {
  closeModal: PropTypes.func,
  files: PropTypes.array,
};

export default GraphicUpload;
