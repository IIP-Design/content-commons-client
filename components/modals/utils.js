import GraphicProject from 'components/GraphicProject/GraphicProject';
import Post from 'components/Post/Post';
import Video from 'components/Video/Video';
import Document from 'components/Document/Document';
/**
 * Check content type of an item and passes in the correct component to the modal
 * @param {Object} item data from the selected item
 */
export const getModalContent = item => {
  const noContent = <div>No content currently available</div>;

  if ( item ) {
    switch ( item.type ) {
      case 'graphic':
        return <GraphicProject item={ item } displayAsModal />;

      case 'video':
        return <Video item={ item } />;

      case 'post':
        return <Post item={ item } />;

      case 'document':
        return <Document item={ item } displayAsModal />;

      default:
        return noContent;
    }
  }

  return noContent;
};
