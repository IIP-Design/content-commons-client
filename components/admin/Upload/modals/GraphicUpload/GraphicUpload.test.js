import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { useFileStateManager } from 'lib/hooks/useFileStateManager';
import { suppressActWarning } from 'lib/utils';

import GraphicUpload from './GraphicUpload';

jest.mock(
  'components/admin/EditFileGrid/EditFileGrid',
  () => function EditFileGrid() { return ''; },
);
jest.mock(
  'components/admin/dropdowns/GraphicStyleDropdown/GraphicStyleDropdown',
  () => function GraphicStyleDropdown() { return ''; },
);
jest.mock(
  'components/admin/Upload/modals/IncludeRequiredFileMsg/IncludeRequiredFileMsg',
  () => function IncludeRequiredFileMsg() { return ''; },
);
jest.mock(
  'components/admin/dropdowns/LanguageDropdown/LanguageDropdown',
  () => function LanguageDropdown() { return ''; },
);
jest.mock(
  'components/admin/dropdowns/SocialPlatformDropdown/SocialPlatformDropdown',
  () => function SocialPlatformDropdown() { return ''; },
);

jest.mock( 'lib/hooks/useFileStateManager' );

jest.mock(
  '@apollo/client',
  () => ( {
    useApolloClient: jest.fn( () => ( {
      writeData: jest.fn(),
    } ) ),
  } ),
);

jest.mock( 'next/router', () => ( { useRouter: jest.fn() } ) );

const files = [
  {
    id: '3db9cba0-4e7d-4a6f-aecf-asdfikq8aa',
    name: '4_3_Serious_FB.jpg',
    size: 433851,
    language: 'ck2lzfx710hkq07206thus6pt',
    style: '',
    social: ['ck9h3m3g626bd07201gh712vk'],
    input: {
      type: 'image/jpeg',
    },
    loaded: 0,
  },
  {
    id: '3db9cba0-4e7d-4a6f-aecf-e8a3e76b3d13',
    name: '4_3_Serious_TW.jpg',
    size: 297343,
    language: 'ck2lzfx710hkq07206thus6pt',
    style: '',
    social: ['ck9h3m3g626bd07201gh712vk'],
    input: {
      type: 'image/jpeg',
    },
    loaded: 0,
  },
  {
    id: '3db9cba0-4e7d-4a6f-aecf-aszia',
    name: 'transcript.docx',
    size: 51893,
    language: 'ck2lzfx710hkq07206thus6pt',
    style: '',
    social: [],
    input: {
      type: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
    },
    loaded: 0,
  },
  {
    id: '52d247e6-ad4a-4f73-a330-372e33ed585b',
    name: 'test-file-shell.png',
    size: 18986,
    language: 'ck2lzfx710hkq07206thus6pt',
    style: 'ck9h3koe426aa0720y421wmk3',
    styleSelection: 'Clean',
    social: [],
    input: {
      type: 'image/png',
    },
    loaded: 0,
  },
  {
    id: '1731fbb4-d785-40d8-a0b5-ad0a542d9d88',
    name: 'OpenSans-Regular.ttf',
    size: 217360,
    language: 'ck2lzfx710hkq07206thus6pt',
    style: '',
    social: [],
    input: {
      type: '',
    },
    loaded: 0,
  },
  {
    id: '2ccc920d-f179-4952-8973-de0770297419',
    name: '4_3_Serious_English-(Simplified)_TW.psd',
    size: 402222,
    language: '',
    style: '',
    social: [],
    input: {
      type: 'image/vnd.adobe.photoshop',
    },
    loaded: 0,
  },
];

describe( '<GraphicUpload />, if graphic files have been selected for upload', () => {
  const props = {
    closeModal: jest.fn(),
    files: [
      {
        lastModified: 1589375838787,
        lastModifiedDate: {},
        name: '4_3_Serious_FB.jpg',
        size: 433851,
        type: 'image/jpeg',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589375841841,
        lastModifiedDate: {},
        name: '4_3_Serious_TW.jpg',
        size: 297343,
        type: 'image/jpeg',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589821475267,
        lastModifiedDate: {},
        name: 'transcript.docx',
        size: 51893,
        type: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589821815569,
        lastModifiedDate: {},
        name: 'test-file-shell.png',
        size: 18986,
        type: 'image/png',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589821518000,
        lastModifiedDate: {},
        name: 'OpenSans-Regular.ttf',
        size: 217360,
        type: '',
        webkitRelativePath: '',
      },
      {
        lastModified: 158187323758173,
        lastModifiedDate: {},
        name: '4_3_Serious_English-(Simplified)_TW.psd',
        size: 402222,
        type: 'image/vnd.adobe.photoshop',
        webkitRelativePath: '',
      },
    ],
  };

  let Component;
  let wrapper;

  useFileStateManager.mockImplementation( () => ( {
    state: { files },
    dispatch: jest.fn(),
  } ) );

  beforeEach( () => {
    Component = <GraphicUpload { ...props } />;
    wrapper = mount( Component );
    // jest.resetModules();
    // useFileStateManager.mockImplementation( () => ( {
    //   state: { files },
    //   dispatch: jest.fn(),
    // } ) );
  } );

  const consoleError = console.error;

  beforeAll( () => suppressActWarning( consoleError ) );
  afterAll( () => {
    console.error = consoleError;
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders the container', () => {
    const container = wrapper.find( '.container' );

    expect( container.exists() ).toEqual( true );
  } );

  it( 'renders the heading', () => {
    const heading = wrapper.find( 'h1' );

    expect( heading.text() ).toEqual( 'SOCIAL MEDIA GRAPHIC PROJECT' );
  } );

  it( 'renders the uploading file count message', () => {
    const filesMsg = wrapper.find( '.files' );

    expect( filesMsg.text() ).toEqual( 'Preparing 6 files for upload...' );
  } );

  it( 'renders the EditFileGrid', () => {
    const editFileGrid = wrapper.find( 'EditFileGrid' );

    expect( toJSON( editFileGrid ) ).toMatchSnapshot();
  } );

  it( 'calling create does not display IncludeRequiredFileMsg', async () => {
    const editFileGrid = wrapper.find( 'EditFileGrid' );
    const requiredFileMsg = () => wrapper.find( 'IncludeRequiredFileMsg' );
    const { onContinue } = editFileGrid.props();

    const createFnTest = async () => {
      await onContinue();
      wrapper.update();
    };

    await createFnTest()
      .then( () => {
        const { includeRequiredFileMsg } = requiredFileMsg().props();

        expect( includeRequiredFileMsg ).toEqual( false );
      } );
  } );
} );

describe( '<GraphicUpload />, if no graphic files have been selected for upload', () => {
  const props = {
    closeModal: jest.fn(),
    files: [
      {
        lastModified: 1589821475267,
        lastModifiedDate: {},
        name: 'transcript.docx',
        size: 51893,
        type: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589821815569,
        lastModifiedDate: {},
        name: 'test-file-shell.png',
        size: 18986,
        type: 'image/png',
        webkitRelativePath: '',
      },
      {
        lastModified: 1589821518000,
        lastModifiedDate: {},
        name: 'OpenSans-Regular.ttf',
        size: 217360,
        type: '',
        webkitRelativePath: '',
      },
      {
        lastModified: 158187323758173,
        lastModifiedDate: {},
        name: '4_3_Serious_English-(Simplified)_TW.psd',
        size: 402222,
        type: 'image/vnd.adobe.photoshop',
        webkitRelativePath: '',
      },
    ],
  };

  let Component;
  let wrapper;

  beforeEach( () => {
    Component = <GraphicUpload { ...props } />;
    wrapper = mount( Component );
    // jest.resetModules();
    useFileStateManager.mockImplementation( () => ( {
      state: {
        files: files.filter( file => {
          const { input: { type }, name } = file;
          const isJpgOrPng = type.includes( 'png' ) || type.includes( 'jpeg' );
          const filename = name.toLowerCase();
          const isFacebook = filename.includes( 'fb' ) || filename.includes( 'facebook' );
          const isTwitter = filename.includes( 'tw' ) || filename.includes( 'twitter' );
          const isGraphicFile = ( isFacebook || isTwitter ) && isJpgOrPng;

          return !isGraphicFile;
        } ),
      },
      dispatch: jest.fn(),
    } ) );
  } );

  const consoleError = console.error;

  beforeAll( () => suppressActWarning( consoleError ) );
  afterAll( () => {
    console.error = consoleError;
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders the uploading file count message', () => {
    const filesMsg = wrapper.find( '.files' );

    expect( filesMsg.text() ).toEqual( 'Preparing 4 files for upload...' );
  } );

  it( 'calling create displays IncludeRequiredFileMsg', async () => {
    const editFileGrid = wrapper.find( 'EditFileGrid' );
    const requiredFileMsg = () => wrapper.find( 'IncludeRequiredFileMsg' );
    const { onContinue } = editFileGrid.props();

    const createFnTest = async () => {
      await onContinue();
      wrapper.update();
    };

    await createFnTest()
      .then( () => {
        const { includeRequiredFileMsg } = requiredFileMsg().props();

        expect( includeRequiredFileMsg ).toEqual( true );
      } );
  } );
} );
