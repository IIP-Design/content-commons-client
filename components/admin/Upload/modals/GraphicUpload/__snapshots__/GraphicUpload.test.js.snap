// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraphicUpload />, if graphic files have been selected for upload renders the EditFileGrid 1`] = `
<EditFileGrid
  allowedFiles=".png,.jpg,.jpeg,.gif,.psd,.ai,.ae,.pdf,.doc,.docx,.ttf,.otf"
  cancelConfirm={true}
  continueBtn="Continue"
  duplicateConfirm={true}
  files={
    Array [
      Object {
        "id": "3db9cba0-4e7d-4a6f-aecf-asdfikq8aa",
        "input": Object {
          "type": "image/jpeg",
        },
        "language": "ck2lzfx710hkq07206thus6pt",
        "loaded": 0,
        "name": "4_3_Serious_FB.jpg",
        "size": 433851,
        "social": Array [
          "ck9h3m3g626bd07201gh712vk",
        ],
        "style": "",
      },
      Object {
        "id": "3db9cba0-4e7d-4a6f-aecf-e8a3e76b3d13",
        "input": Object {
          "type": "image/jpeg",
        },
        "language": "ck2lzfx710hkq07206thus6pt",
        "loaded": 0,
        "name": "4_3_Serious_TW.jpg",
        "size": 297343,
        "social": Array [
          "ck9h3m3g626bd07201gh712vk",
        ],
        "style": "",
      },
      Object {
        "id": "3db9cba0-4e7d-4a6f-aecf-aszia",
        "input": Object {
          "type": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
        },
        "language": "ck2lzfx710hkq07206thus6pt",
        "loaded": 0,
        "name": "transcript.docx",
        "size": 51893,
        "social": Array [],
        "style": "",
      },
      Object {
        "id": "52d247e6-ad4a-4f73-a330-372e33ed585b",
        "input": Object {
          "type": "image/png",
        },
        "language": "ck2lzfx710hkq07206thus6pt",
        "loaded": 0,
        "name": "test-file-shell.png",
        "size": 18986,
        "social": Array [],
        "style": "ck9h3koe426aa0720y421wmk3",
        "styleSelection": "Clean",
      },
      Object {
        "id": "1731fbb4-d785-40d8-a0b5-ad0a542d9d88",
        "input": Object {
          "type": "",
        },
        "language": "ck2lzfx710hkq07206thus6pt",
        "loaded": 0,
        "name": "OpenSans-Regular.ttf",
        "size": 217360,
        "social": Array [],
        "style": "",
      },
      Object {
        "id": "2ccc920d-f179-4952-8973-de0770297419",
        "input": Object {
          "type": "image/vnd.adobe.photoshop",
        },
        "language": "",
        "loaded": 0,
        "name": "4_3_Serious_English-(Simplified)_TW.psd",
        "size": 402222,
        "social": Array [],
        "style": "",
      },
    ]
  }
  onAdd={[Function]}
  onCancel={[MockFunction]}
  onContinue={[Function]}
  onRemove={[Function]}
  onUpdate={[Function]}
  removeConfirm={true}
  screens={
    Array [
      Array [
        Object {
          "allowedFiles": ".png,.jpg,.jpeg,.gif,.pdf,.doc,.docx,",
          "component": [Function],
          "label": "Language",
          "name": "language",
          "props": Object {
            "search": true,
          },
        },
      ],
      Array [
        Object {
          "allowedFiles": ".png,.jpg,.jpeg,.gif",
          "component": [Function],
          "label": "Style",
          "name": "style",
        },
        Object {
          "allowedFiles": ".png,.jpg,.jpeg,.gif",
          "component": [Function],
          "label": "Platform",
          "name": "social",
          "props": Object {
            "closeOnChange": true,
          },
        },
      ],
    ]
  }
  showCompletionTracker={true}
  showHeader={true}
/>
`;
